docs/standard_library/slice_methods.md

```diff
@@ -0,0 +1,193 @@
+---
+title: Slice Methods
+description:
+  Learn about the commonly used methods available for slices in Noir, including push_back, len, srt, map, fold, reduce, all, and any.
+keywords: [rust, slice, methods, push_back, len, sort, fold, reduce, all, any]
+---
+
+# Slice
+
+For convenience, the STD provides some ready-to-use, common methods for slices:
+
+## push_back
+
+Pushes a new element to the end of the slice, returning a new slice with a length one greater than the original unmodified slice.
+
+```rust
+fn push_back<T>(_self: [T], _elem: T) -> [T]
+```
+
+example:
+
+```rust
+fn main() -> pub Field {
+    let mut slice: [Field] = [0; 2];
+
+    let mut new_slice = slice.push_back(6);
+    new_slice.len()
+}
+```
+
+## len
+
+Returns the length of a slice
+
+```rust
+fn len<T>(_slice: [T]) -> comptime Field
+```
+
+example
+
+```rust
+fn main() {
+    let slic = [42, 42]
+    assert(slic.len() == 2);
+}
+```
+
+## sort
+
+Returns a new sorted slice. The original slice remains untouched. Notice that this function will
+only work for slices of fields or integers, not for any arbitrary type. This is because the sorting
+logic it uses internally is optimized specifically for these values. If you need a sort function to
```

Comment:
```suggestion
only work for slices of fields or integers, not for any arbitrary type. This is because the sorting
logic the function uses internally is optimized specifically for these values. If you need a sort function to
```